### PR TITLE
Add v8::Object::HasOwnProperty bindings

### PR DESCRIPTION
**Pull request by littledivy**

### CODE REVIEW

**Comment by littledivy** [comment]
Needed for https://github.com/denoland/deno/pull/13633

File: src/binding.cc [hunk 1]
```diff
@@ -1158,6 +1158,12 @@ MaybeBool v8__Object__HasIndex(const v8::Object& self,
       ptr_to_local(&self)->Has(ptr_to_local(&context), index));
 }
 
+MaybeBool v8__Object__HasOwnProperty(const v8::Object& self, const v8::Context& context,
+                          const v8::Value& key) {
+  return maybe_to_maybe_bool(
+      ptr_to_local(&self)->HasOwnProperty(ptr_to_local(&context), ptr_to_local(&key).As<v8::Name>()));
```
Comment (piscisaureus) [inline src/binding.cc:1164]: This cast is not safe I think.

File: src/binding.cc [hunk 2]
```diff
@@ -1158,6 +1158,12 @@ MaybeBool v8__Object__HasIndex(const v8::Object& self,
       ptr_to_local(&self)->Has(ptr_to_local(&context), index));
 }
 
+MaybeBool v8__Object__HasOwnProperty(const v8::Object& self, const v8::Context& context,
+                          const v8::Value& key) {
```
Comment (piscisaureus) [inline src/binding.cc:1162]: Follow the C++ API unless there are overriding concerns.

```suggestion
                          const v8::Name& key) {
```

File: src/object.rs [hunk 3]
```diff
@@ -99,6 +99,11 @@ extern "C" {
     context: *const Context,
     index: u32,
   ) -> MaybeBool;
+  fn v8__Object__HasOwnProperty(
+    this: *const Object,
+    context: *const Context,
+    key: *const Value,
```
Comment (piscisaureus) [inline src/object.rs:105]:
```suggestion
    key: *const Name,
```

File: tests/test_api.rs [hunk 4]
```diff
@@ -1550,6 +1550,7 @@ fn object() {
     assert_ne!(id, 0);
 
     assert!(object.has(scope, n1.into()).unwrap());
+    assert!(object.has_own_property(scope, n1.into()).unwrap());
```
Comment (piscisaureus) [inline tests/test_api.rs:1553]: Also add a test where `has_own_property()` returns false.

**Comment by piscisaureus** [comment]
LGTM

**Review by piscisaureus — APPROVED** [review]

**Pull request merged**